pd/src/pending_block.rs

```diff
@@ -61,19 +72,37 @@ impl PendingBlock {
 
     /// Adds the state changes from a verified transaction.
     pub fn add_transaction(&mut self, transaction: VerifiedTransaction) {
+        let no_undelegations = transaction.undelegation_validators.is_empty();
```

Comment:
Currently this is just used in processing the notes, but it should also be used in processing nullifiers below, something like

```rust
for nullifier in transaction.spent_nullifiers {
    self.spent_nullifiers.insert(nullifier);
    if undelegations {
        self.quarantined_nullifiers.do_stuff();
    }
}
```

Since the nullifier processing path is "do A unconditionally, if undelegations, do B", while the note processing path is "do A if no undelegations, do B if undelegations", I think it'd be clearer if this bool were changed to
```rust
        let undelegations = !transaction.undelegation_validators.is_empty();
```

Reply:
done!